### PR TITLE
V3.6.0 particle material switch error fix

### PR DESCRIPTION
**Pull request by zxx43**
Re: #

### Changelog

* 

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.

  > Manual trigger with `@cocos-robot run test cases` afterward.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Comment by zxx43** [comment]
@cocos-robot run test cases

File: cocos/particle/renderer/particle-system-renderer-data.ts [hunk 1]
```diff
@@ -151,12 +152,67 @@ export default class ParticleSystemRenderer {
...
+    public get cpuMaterial () {
+        return this._cpuMaterial;
+    }
+
+    public set cpuMaterial (val: Material | null) {
+        if (val === null) {
+            return;
+        } else {
+            const effectName = val.effectName;
+            if (effectName.indexOf('particle') === -1 || effectName.indexOf('particle-gpu') !== -1) {
```
Comment (SantyWang) [inline cocos/particle/renderer/particle-system-renderer-data.ts:172]: Why check "particle-gpu" ?

**Comment by AndyYuTian** [comment]
At the current stage, this change will lead to all users previous made particles system lost their material, which is unacceptable for users. it should according to the user existing setup, put users' material into the right property. Plus, the user interface doesn't update immediately after user update the useGPU settings, which does not supposed to happen as well.

File: cocos/particle/renderer/particle-system-renderer-data.ts (hunk 1 above)
Comment (zxx43) [reply cocos/particle/renderer/particle-system-renderer-data.ts:171]: do not use effect include gpu for cpu material

Comment (zxx43) [reply cocos/particle/renderer/particle-system-renderer-data.ts:172]: > Why check "particle-gpu" ?

do not use effect include gpu for cpu material

**Comment by zxx43** [comment]
> At the current stage, this change will lead to all users previous made particles system lost their material, which is unacceptable for users. it should according to the user existing setup, put users' material into the right property. Plus, the user interface doesn't update immediately after user update the useGPU settings, which does not supposed to happen as well.

fixed

**Comment by LIUYIa** [comment]
@cocos-robot run test cases

**Review by SantyWang — APPROVED** [review]

**Review by SantyWang — APPROVED** [review]

**Pull request merged**